src/components/AddCardContainer.tsx

```diff
@@ -0,0 +1,287 @@
+import React, { ChangeEvent, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+import CardPreview from './CardPreview';
+import Input from './Input';
+import { useCardDispatch } from '../context/CardContext';
+import { isAlphabet, isNumber, regEx, validateMonth, validateYear } from '../utils/validateInput';
+import ErrorMessage from './ErrorMessage';
+import { getUniqueID } from '../utils/key';
+
+const AddCardContainer = () => {
+  const [cardNumbers, setCardNumbers] = useState<string[]>(['', '', '', '']);
+  const [cardExpirationDate, setCardExpirationDate] = useState<string[]>(['', '']);
+  const [cardOwner, setCardOwner] = useState<string[]>(['']);
+  const [cardCVC, setCardCVC] = useState<string[]>(['']);
+  const [cardPWD, setCardPWD] = useState<string[]>(['', '']);
+  const [expirationError, setExpirationError] = useState<boolean>(false);
+  const setCard = useCardDispatch();
+  const navigate = useNavigate();
+
+  const onChangeDateState = (e: ChangeEvent) => {
+    if (!(e.target instanceof HTMLInputElement)) return;
+
+    switch (e.target.dataset.id) {
+      case '0':
+        validateMonth(e.target);
+        break;
+      case '1':
+        validateYear(e.target);
+        break;
+      default:
+    }
+
+    onChangeState(cardExpirationDate, setCardExpirationDate, 'number')(e);
+  };
+
+  const onChangeState =
+    (state: string[], setState: React.Dispatch<React.SetStateAction<string[]>>, type: string) =>
+    (e: ChangeEvent) => {
+      if (!(e.target instanceof HTMLInputElement)) return;
+
+      switch (type) {
+        case 'number':
+        case 'password':
+          if (!isNumber(e.target.value))
+            e.target.value = e.target.value.replace(regEx.notNumber, '');
+          break;
+        case 'text':
+          if (!isAlphabet(e.target.value))
+            e.target.value = e.target.value.replace(regEx.notAlphbet, '');
+          break;
+        default:
+      }
+
+      const copyState = [...state];
+      copyState[Number(e.target.dataset.id)] = e.target.value;
+      setState(copyState);
+    };
```

Comment:
위의 코멘트와 동일한 맥락인데,
기존에 선언된 참조값의 속성값을 변경하는 패턴은 지양하는 것이 좋습니다.
이런식으로 값이 변경된 새로운 참조값을 선언해주세요.


```suggestion
      const targetId = Number(e.target.dataset.id);
      const nextState = state.map((val, index) => index === targetId ? e.target.value : val);
      setState(nextState);
    };
```